lib/rucio/core/rule.py

```diff
@@ -2414,22 +2414,24 @@ def examine_rule(rule_id: str, *, session: "Session") -> dict[str, Any]:
 
 
 @transactional_session
-def get_evaluation_backlog(expiration_time: int = 600, *, session: "Session") -> tuple[int, datetime]:
+def get_evaluation_backlog(expiration_time: int = 600, *, session: "Session") -> Union[NoValue, tuple[int, datetime]]:
     """
     Counts the number of entries in the rule evaluation backlog.
     (Number of files to be evaluated)
 
     :returns:     Tuple (Count, Datetime of oldest entry)
     """
 
-    cached_backlog = REGION.get('rule_evaluation_backlog', expiration_time=expiration_time)
+    cached_backlog: Union[NoValue, tuple[int, datetime]] = REGION.get('rule_evaluation_backlog', expiration_time=expiration_time)
     if cached_backlog is NO_VALUE:
-        stmt = select(func.count(models.UpdatedDID.created_at), func.min(models.UpdatedDID.created_at))
-        result = session.execute(stmt).scalars().one()
+        stmt = select(
+            func.count(models.UpdatedDID.created_at),
+            func.min(models.UpdatedDID.created_at)
+        )
+        result = session.execute(stmt).one()._tuple()
         REGION.set('rule_evaluation_backlog', result)
         return result
-    else:
-        return cached_backlog  # type: ignore
+    return cached_backlog
```

Comment:
I’m not asking you to change this, but I am wondering why linters think omitting the `else` clause is preferable.

Reply:
It wasn't the linter - I omitted it because it was omitted in the changes you linked to as guidelines for this change: https://github.com/rucio/rucio/pull/6493/files#r1521715742

I'm okay with either having it removed or keeping it

Reply:
No, please keep it. I guess that @erlingstaff did it because of [R1705](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html). I just can’t agree that it is *undeniably* better than the ‘problematic’ code.